### DISCUSSION TITLE
Temp refetchInterval

### DISCUSSION THREAD

**Discussion by romansndlr** [TanStack/query#1083] [General]
Hey all, I work in an event-sourced environment, and a lot of the time we need to wait for all the async messages to be registered before we can redirect the user. I would like to have a possibility in react-query to set up a temporary refect interval. So basically adding the ability to pass milliseconds to this property that define for how long we want to run this "poling" from the API.

**Comment by TkDodo** [comment]
Have you had a look at the [auto refetching](https://react-query.tanstack.com/docs/examples/auto-refetching) example?

if you set `refetchInterval` to `false`, the polling will stop. If you want to have polling enabled every second, but only for like 10 seconds, I would suggest doing a custom `useEffect` with a `setTimeout` that will then after 10 seconds set a state value to `false`, which will trigger a re-render and thus turn off refetching.

does that make sense?
